### PR TITLE
Add {{.ID}} to podman network ls.

### PR DESCRIPTION
**Pull request by eiffel-fl**
Hi.

I managed to add creation of ID with stringid.GenerateNonCryptoID() in NewNcList.
I also add the ID to NetworkConfigList and the ID is read from the file in ConfListFromBytes.
So the (long) ID are displayed when using `./bin/podman network ls --format '{{.ID}}'`.

Even if it works I have the feeling that I forgot to modify some parts of the code especially the code related to function buildOneConfig.
Moreover, I had to modify code in libcni so maybe the library has to take into account ID before implementing that.

Maybe I can just slightly modify podman code to print something like "This will be implemented in near future" instead of network ID when calling `./bin/podman network ls --format '{{.ID}}'`?

Best regards.

### CODE REVIEW

**Comment by zhangguanzhang** [comment]
You cannot change the vendor files directly, `vendor/*` is managed by go command, not by user to edit them

**Comment by eiffel-fl** [comment]
I understand and this is why this PR will not be merged.
Moreover, @mheon told me that this is complicated to add ID to networks but he also adviced me to submit what I wrote.
Maybe this code can be used later.

**Comment by rhatdan** [comment]
Crazy idea but could we just return `name` when the user asks for ID?  Is the issue here compatibility with Docker?

**Comment by eiffel-fl** [comment]
I though of printing "Will be implemented later" instead of ID so printing the name seems good for me.
If by compatibility issue with Docker you mean different output yes it is:
```bash
$ docker network create bar
$ docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
277bf869b449        bar                 bridge              local
0c05a2190d63        bridge              bridge              local
b178ed94c7c5        host                host                local
30913563a58e        none                null                local
```
```bash
$ podman network create foo
$ podman network ls
NAME    VERSION  PLUGINS
foo     0.4.0    bridge,portmap,firewall,dnsname
```

But I think the most important thing is that `podman` exits when ID is given in format (see #7865):
```bash
$ podman network ls --format '{{.ID}}'
Error: template: listNetworks:1:14: executing "listNetworks" at <.ID>: can't evaluate field ID in type network.ListPrintReport
```

**Comment by mheon** [comment]
I don't think we can necessarily just send the name again, but sending the SHA256 of the name seems like it would be 100% compatible with what Docker is doing and low-effort.

**Comment by rhatdan** [comment]
Is the end goal here ot make sure the APIV2 bindings also return an ID so that scripts that use it in docker-py/dockercompose will work?

**Comment by eiffel-fl** [comment]
@mheon what you proposed is a good idea so I implemented that, go format are really powerful!

**Comment by rhatdan** [comment]
@eiffel-fl What is this goal of this PR?  What is the issue you are trying to solve?  Is the problem that users see a diffent display when using Podman versus Docker?  Or is the tools that are scraping the output of podman network ls, that are breaking?

**Comment by eiffel-fl** [comment]
@rhatdan My goal is to fix the issue #7865 (sorry I forgot to mention it in PR header).
This issue shows that `podman` returns an error when using `{{.ID}}` inside a format.
I only want `podman` to returns without error when this is used as format.

The problem is that I can either add ID to network but this seems impossible to do now because I needed to modify some code related to `libcni` which is in the `vendor` directory.
So, I decided to implement what @mheon proposed because this seems to be a good trade-off because it solves the issue problem and does not need modifications to code inside `vendor/`.

File: pkg/network/netconflist.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"net"
 	"os"
 	"path/filepath"
+	"github.com/containers/storage/pkg/stringid"
```
[the remark this answers is not in the document]
Comment (eiffel-fl) [reply pkg/network/netconflist.go]: Sorry I forgot to run `make gofmt` on my code.
But normally with my last modifications I did not touch to this file so this modification is now null and void.

**Comment by rhatdan** [comment]
@eiffel-fl Still working on this one?

**Comment by eiffel-fl** [comment]
@rhatdan Not really, I do not really know what to do more than what I did since the discussion was not really concluded.
But if you see another way to do it, I would be happy to work it!

**Comment by rhatdan** [comment]
@mheon @baude What do you want to do with this one?

**Comment by mheon** [comment]
Probably can't be done until we do a more thorough rework of networks

**Comment by eiffel-fl** [comment]
For now I will let this merge request as draft.
Please let me know when someone reworks the networks so I can try to add this contribution.

**Comment by Luap99** [comment]
Thanks @eiffel-fl for you contribution. I have added network ID support in #8549. This also includes the ID field support for podman network ls, so we can close this.

**Comment by eiffel-fl** [comment]
@Luap99 You are welcome. I looked at #8549 and your modifications are cleaner than what I proposed so it is better for the code base!

**Pull request closed without merging**